src/js/components/Select/Select.js

```diff
@@ -284,6 +285,11 @@ const Select = forwardRef(
 
     const iconColor = getIconColor(theme);
 
+    let ariaLabelledBy;
+    if (formContext?.useFormField({})?.inForm && id && !ariaLabel) {
+      ariaLabelledBy = `${id}__input__label`;
```

Comment:
I think we also wanted to try including the original id, see comment https://github.com/grommet/grommet/pull/7369#discussion_r1815792748

Reply:
added